src/backend/marsha/core/simple_jwt/tokens.py

```diff
@@ -15,6 +15,47 @@
 from .permissions import ResourceAccessPermissions
 
 
+class MarshaRefreshToken(RefreshToken):
+    """
+    Refresh token made especially for Marsha. In this token, we also set the access token type.
+    We will use later when the access token is generated. Instead of picking the token type from
+    the access token class, we will use the one set in the refresh token payload. This avoid to
+    have a view/serializer for each refresh token type we are managing.
+    """
```

Comment:
Comment may explain the `token_type` attribute must not be overridden then. 

Actually I don't like the fact that we can mix use of `ResourceRefreshToken` and `UserRefreshToken`, but for sure it allows to easily have only one endpoint (otherwise the `UserTokenRefreshSerializer` or its corresponding view would have to be a bit more complex to detect refresh token's type).

``` python
resource_refresh_jwt = ResourceRefreshToken()
assert UserRefreshToken(resource_refresh_jwt)  # works
```